crates/miden-tx/src/executor/data_store.rs

```diff
@@ -27,11 +28,11 @@ pub trait DataStore: MastForestStore {
     /// - The block with the specified number could not be found in the data store.
     /// - The combination of specified inputs resulted in a transaction input error.
     /// - The data store encountered some internal error
-    fn get_transaction_inputs(
+    fn get_transaction_preparation_inputs(
         &self,
         account_id: AccountId,
         ref_blocks: BTreeSet<BlockNumber>,
-    ) -> impl FutureMaybeSend<Result<(PartialAccount, BlockHeader, PartialBlockchain), DataStoreError>>;
+    ) -> impl FutureMaybeSend<Result<TransactionPreparationInputs, DataStoreError>>;
```

Comment:
Assuming the above comments work out, the value of introducing this extra type here seems relatively low to me - so, I'd revert this change.